power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -53,14 +199,64 @@ inline void get_component_sequence(MainModelState<ComponentContainer> const& sta
 template <component_c Component, class ComponentContainer,
           forward_iterator_like<typename Component::UpdateType> ForwardIterator>
     requires model_component_state_c<MainModelState, ComponentContainer, Component>
-inline std::vector<Idx2D> get_component_sequence(MainModelState<ComponentContainer> const& state, ForwardIterator begin,
-                                                 ForwardIterator end, Idx n_comp_elements = na_Idx) {
+inline std::vector<Idx2D> get_component_sequence_by_iter(MainModelState<ComponentContainer> const& state,
+                                                         ForwardIterator begin, ForwardIterator end,
+                                                         Idx n_comp_elements = na_Idx) {
     std::vector<Idx2D> result;
     result.reserve(std::distance(begin, end));
-    get_component_sequence<Component>(state, begin, end, std::back_inserter(result), n_comp_elements);
+    get_component_sequence_impl<Component>(state, begin, end, std::back_inserter(result), n_comp_elements);
     return result;
 }
 
+// get sequence idx map of a certain batch scenario
+template <typename CompType, class ComponentContainer>
+std::vector<Idx2D> get_component_sequence(MainModelState<ComponentContainer> const& state,
+                                          ConstDataset const& update_data, Idx scenario_idx,
+                                          independence::UpdateCompProperties const& comp_independence = {}) {
+    auto const get_sequence = [&state, n_comp_elements = comp_independence.get_n_elements()](auto const& span) {
+        return get_component_sequence_by_iter<CompType>(state, std::begin(span), std::end(span), n_comp_elements);
+    };
+    if (update_data.is_columnar(CompType::name)) {
+        auto const buffer_span =
+            update_data.get_columnar_buffer_span<meta_data::update_getter_s, CompType>(scenario_idx);
+        return get_sequence(buffer_span);
+    }
+    auto const buffer_span = update_data.get_buffer_span<meta_data::update_getter_s, CompType>(scenario_idx);
+    return get_sequence(buffer_span);
+}
+} // namespace detail
+
+template <class... ComponentTypes, class ComponentContainer>
+SequenceIdx<ComponentTypes...> get_all_sequence_idx_map(MainModelState<ComponentContainer> const& state,
```

Comment:
I kinda wanted to mention this as well but hesitated because of the scope. It's a sequence of `Idx2D`, since we do have `idx`, to be more clear, all this chain of functions should be named `get_*Idx2D_sequence`.

Reply:
https://github.com/PowerGridModel/power-grid-model/issues/839

Reply:
> #839

Resolving due to this.